### PR TITLE
Add more unittest 

### PR DESCRIPTION
**Pull request by pan-x-c**
1. setup local actions runner on a GPU machine
2. use docker-compose to setup a cluster
3. add more unit tests
   - single process
   - multi process
   - multi process with GPU
   - ray with CPU
   - ray with GPU

### CODE REVIEW

File: .github/workflows/docker/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    deploy:
+      resources:
+        reservations:
+          devices:
+          - driver: nvidia
+            device_ids: ['0', '1']
+            capabilities: [gpu]
+
+  ray-worker:
+    image: data-juicer-unittest:0.2.0
```
Comment (yxdyc) [inline .github/workflows/docker/docker-compose.yml:33]: In the future, we may change it into `data-juicer-unittest:latest` and maintain a latest image

File: data_juicer/utils/unittest_utils.py [hunk 2]
```diff
@@ -2,11 +2,30 @@
...
+from datasets import Dataset
+
+from data_juicer.ops import Filter
+from data_juicer.utils.constant import Fields
 from data_juicer.utils.registry import Registry
 
 SKIPPED_TESTS = Registry('SkippedTests')
 
 
+def TEST_TAG(*tags):
```
Comment (yxdyc) [inline data_juicer/utils/unittest_utils.py:17]: There are two type of tags, while the PR's description mentioned 5 types of test scenarios. So what is the mapping relationship? 
```
single process
multi process
multi process with GPU
ray with CPU
ray with GPU
```

File: .github/workflows/unittest.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  unittest-single:
+    runs-on: [self-hosted, linux]
```
Comment (yxdyc) [inline .github/workflows/unittest.yml:18]: By now, does our self-hosted machine satisfy the requirements from security colleagues?

File: data_juicer/utils/unittest_utils.py (hunk 2 above)
Comment (pan-x-c) [reply data_juicer/utils/unittest_utils.py:17]: For now, there is no distinction between CPU and GPU testing, and the corresponding configuration will be added in the future when needed.

File: .github/workflows/unittest.yml (hunk 3 above)
Comment (pan-x-c) [reply .github/workflows/unittest.yml:18]: Nope

**Review by drcege — APPROVED** [review]

**Pull request merged**